pkg/metrics/statsd/statsd.go

```diff
@@ -29,19 +31,61 @@ type statsd struct {
 	conn net.Conn
 	ch   chan *metric
 
+	// extraDimensions are values added to every emit (e.g. location)
+	extraDimensions map[string]string
+
 	now func() time.Time
 }
 
 // New returns a new metrics.Emitter
-func New(ctx context.Context, log *logrus.Entry, env env.Core, account, namespace string, mdmSocketEnv string) metrics.Emitter {
+func New(ctx context.Context, env env.Core, account, namespace string, mdmSocketEnv string) metrics.Emitter {
 	s := &statsd{
-		log: log,
+		log: env.LoggerForComponent("metrics"),
 		env: env,
 
 		account:      account,
 		namespace:    namespace,
 		mdmSocketEnv: mdmSocketEnv,
 
+		extraDimensions: map[string]string{
+			"hostname": env.Hostname(),
+			"location": env.Location(),
+			"service":  env.Service(),
+			"version":  version.GitCommit,
+		},
+
+		ch: make(chan *metric, 1024),
+
+		now: time.Now,
+	}
+
+	if s.account == "" {
+		s.account = "*"
+	}
+
+	if s.namespace == "" {
+		s.namespace = "*"
+	}
+
+	go s.run()
```

Comment:
Having the run started implicitly inside of the new is introducing potential side effects - I'd prefer it at all possible for the goroutine to be managed with the component that has the lifetime of the metricsEmitter

Reply:
fixed